third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp

```diff
@@ -56,20 +56,34 @@ std::optional<int64_t> getFoldedConstantValue(Operation *op) {
     return intAttr.value();
   }
 
-  auto val = cast<Value>(results[0]);
+  auto val = dyn_cast<Value>(results[0]);
+  assert(val && "op must have at least one result");
   auto constOp = val.getDefiningOp<arith::ConstantOp>();
   if (!constOp)
     return std::nullopt;
 
-  return getIntAttr(constOp.getValue());
+  intAttr = getIntAttr(constOp.getValue());
+  if (intAttr.has_value()) {
+    return intAttr.value();
+  }
+  return std::nullopt;
 }
 
-// return true if the `val` value is a constant containing a value equal to zero
-bool hasConstZero(Value val) {
-  auto intVal = getFoldedConstantValue(val.getDefiningOp());
-  return (intVal.has_value() && (intVal.value() == 0));
+// return true if the `val` value is a constant containing a value equal to ref
+bool hasConstEqualTo(const Value val, const int ref) {
+  auto defOp = val.getDefiningOp();
+  if (!defOp)
+    return false;
+  auto intVal = getFoldedConstantValue(defOp);
+  if (intVal.has_value()) {
+    return (intVal.value() == ref);
+  }
+  return false;
```

Comment:
```suggestion
  return getFoldedConstantValue(defOp) == ref;
```
Use `bool operator==(const std::optional<T> &, const T &);`

Reply:
Thanks for the suggestion. The code has been updated in this way.